src/Phinx/Db/Adapter/AdapterInterface.php

```diff
@@ -417,31 +417,31 @@ public function hasPrimaryKey($tableName, $columns, $constraint = null);
      * @param string|null $constraint Constraint name
      * @return bool
      */
-    public function hasForeignKey($tableName, $columns, $constraint = null);
+    public function hasForeignKey(string $tableName, $columns, ?string $constraint = null): bool;
 
     /**
      * Returns an array of the supported Phinx column types.
      *
      * @return string[]
      */
-    public function getColumnTypes();
+    public function getColumnTypes(): array;
 
     /**
      * Checks that the given column is of a supported type.
      *
      * @param \Phinx\Db\Table\Column $column Column
      * @return bool
      */
-    public function isValidColumnType(Column $column);
+    public function isValidColumnType(Column $column): bool;
 
     /**
      * Converts the Phinx logical type to the adapter's SQL type.
      *
-     * @param string $type Type
-     * @param int|null $limit Limit
+     * @param string|\Phinx\Util\Literal $type Type
+     * @param float|null $limit Limit
      * @return array
      */
-    public function getSqlType($type, $limit = null);
+    public function getSqlType($type, ?float $limit = null): array;
```

Comment:
This was changed from `int` to `float` to accommodate the `MysqlAdapter::INT_BIG` constant which has a value of `18446744073709551615` where that is treated as a `float` within PHP. Typing this (or having the integer typedoc) is then technically incorrect, and would throw a type error if we were to turn on `declare(strict_types=1);` and then `MysqlAdapter::INT_BIG` was used. The best option perhaps would be to type as `float`, and then just document that while it's a `float`, only integer like numbers should be used and decimal values will not work. The problem though is that now integers passed into the function are implicitly converted to floats (e.g. `4` -> `4.0`) which is a bit odd.

Could also maybe insert a:
```
if ($limit <= PHP_INT_MAX) {
    $limit = (int)$limit;
}
```
in the implemented functions to retain the current behavior best we can while having proper typehints?

Reply:
This was reverted, such that this function expects an integer again.